server/services/store/sqlstore/board.go

```diff
@@ -642,24 +642,86 @@ func (s *SQLStore) getMemberForBoard(db sq.BaseRunner, boardID, userID string) (
 	return members[0], nil
 }
 
+func (s *SQLStore) implicitBoardMembershipsFromRows(rows *sql.Rows) ([]*model.BoardMember, error) {
+	boardMembers := []*model.BoardMember{}
+
+	for rows.Next() {
+		var boardMember model.BoardMember
+
+		err := rows.Scan(
+			&boardMember.UserID,
+			&boardMember.BoardID,
+		)
+		if err != nil {
+			return nil, err
+		}
+		boardMember.Roles = "editor"
+		boardMember.SchemeEditor = true
+		boardMember.Synthetic = true
+
+		boardMembers = append(boardMembers, &boardMember)
+	}
+
+	return boardMembers, nil
+}
+
 func (s *SQLStore) getMembersForUser(db sq.BaseRunner, userID string) ([]*model.BoardMember, error) {
 	query := s.getQueryBuilder(db).
 		Select(boardMemberFields...).
 		From(s.tablePrefix + "board_members AS BM").
 		LeftJoin(s.tablePrefix + "boards AS B ON B.id=BM.board_id").
 		Where(sq.Eq{"BM.user_id": userID})
 
-	rows, err := query.Query()
+	explicitMemberRows, err := query.Query()
+	if err != nil {
+		s.logger.Error(`getMembersForUser ERROR`, mlog.Err(err))
+		return nil, err
+	}
+	defer s.CloseRows(explicitMemberRows)
+
+	explicitMembers, err := s.boardMembersFromRows(explicitMemberRows)
+	if err != nil {
+		s.logger.Error(`getMembersForUser ERROR`, mlog.Err(err))
+		return nil, err
+	}
+
+	user, err := s.GetUserByID(userID)
+	if err != nil {
+		return nil, err
+	}
+	if user.IsGuest {
+		return explicitMembers, nil
+	}
+
+	newQuery := s.getQueryBuilder(db).
```

Comment:
Could you change this name to something more descriptive like `implicitMembersQuery`?